gef-ui/server/index.js

```diff
@@ -82,7 +82,23 @@ app.set('trustproxy', true);
 app.use(session(sessionOptions));
 app.use(cookieParser());
 app.use(express.urlencoded({ extended: true }));
-app.use('/', supportingDocuments);
+
+// These routes cannot use the csrf check below so must come before it
+// They implement their own separate csrf check
+app.use('/', supportingInformationUploadRoutes);
+
+// For endpoints requiring file uploads, the csrf token can only be sent as part of the query rather than in the body.
+// This is because these forms must use enctype="multipart/form-data"
+// The csrf middleware below expects to find the token in the body, so we copy it from the query to the body here.
+const copyCsrfTokenFromQueryToBody = (req, res, next) => {
```

Comment:
Please move this middleware to `middleware` directory as a function.

Reply:
This middleware has been added as Global middleware therefore effecting all the routes, has this caused other routes to fail?

Reply:
From what I can tell, I don't think that it has caused any of the other routes to fail.
 The e2e tests all still seem to pass too so I think it should be safe